### PR TITLE
issue #55 bugfix for wrong error code on specified specs parameter

### LINKED ISSUE #55: Wrong exit code with '--specs' parameter
When running `cplace-cli` with the `--specs` parameter, the program will show an error and exit with exit code '1' although all tests that are met by the filter have passed.

Example:
`cplace-cli e2e --specs '**/login.spec.ts' && echo 'It works correctly if we see this text'`

Output:
> [...]
> One of the E2E Tests in plugin cf.cplace.training.extended has failed
> Could not execute given command:
>         One of the E2E Tests has failed

This is especially problematic on continous integration environments. It would mark the tests as failed, although all tests passed.

### PR DESCRIPTION
**Pull request by stefancplace**
Fixes #55 

changelog: bugfix for wrong error code on specified specs parameter [Issue cplace-npm-tools#55, PR cplace-npm-tools#60]

### CODE REVIEW

File: src/commands/e2e/TestRunner.ts [hunk 1]
```diff
@@ -1,37 +1,48 @@
...
         const launcherModule = require(this.getWdioCliExecutable());
         let hasFailedTest: boolean = false;
         for (const plugin of this.plugins) {
-            const wdioConf = path.join(this.workingDir, plugin, 'assets', 'e2e', WdioConfigGenerator.WDIO_CONF_NAME);
-            const launcher = new launcherModule.default(wdioConf, {args: ['']});
-            const testHasFailed: boolean = await launcher.run();
-            if (testHasFailed) {
-                console.warn(`One of the E2E Tests in plugin ${plugin} has failed`);
-                hasFailedTest = true;
+            if (this.specsParameter && this.isSpecInPlugin(plugin) || !this.specsParameter) {
```
Comment (rose-m) [inline src/commands/e2e/TestRunner.ts:26]: Please add `( )` to explicitly state order of evaluation of the `&&` and `||` expressions :)

File: src/commands/e2e/util.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import * as path from 'path';
+
+export const getPathToE2E = (workingDir: string, plugin: string): string => {
```
Comment (rose-m) [inline src/commands/e2e/util.ts:3]: Please just export them as `functions` instead of inline lambda constants :)

**Review by rose-m — CHANGES_REQUESTED** [review]
See comments - apart from that looks good :)

**Review by rose-m — APPROVED** [review]
Didn't test but LGTM! 👍

**Review by stophi-dev — APPROVED** [review]
Looks good. 
I couldn't test it right now. But if required, I could do this later

**Comment by stefancplace** [comment]
@stophi-dev always good if somebody else takes a look, I will merge tomorrow then.

**Comment by stophi-dev** [comment]
I tested with this version the following commands:
` cplace-cli e2e --headless --specs '**' && echo 'It works correctly if we see this text'` 
`cplace-cli e2e --specs '**/login.spec.ts' && echo 'It works correctly if we see this text'`
Both passed as expected and didn't return an error code (as before).

Also, 
`cplace-cli e2e --specs '**' --plugins cf.cplace.platform && echo 'It works correctly if we see this text'`
still works (as it did before) and does not return an error code.

**Pull request merged**